packages/scheduled-tasks/src/lib/strategies/ScheduledTaskSQSStrategy.ts

```diff
@@ -0,0 +1,89 @@
+import { container } from '@sapphire/framework';
+import type { SQS } from 'aws-sdk';
+import { randomBytes } from 'crypto';
+import { Consumer, ConsumerOptions } from 'sqs-consumer';
+import { Producer } from 'sqs-producer';
+import type { ScheduledTaskCreateRepeatedTask, ScheduledTasksTaskOptions } from '../types';
+import type { ScheduledTaskBaseStrategy } from '../types/ScheduledTaskBaseStrategy';
+
+export interface ScheduledTaskSQSStrategyMessageBody {
+	task: string;
+	payload?: unknown;
+	options: ScheduledTasksTaskOptions;
+}
+
+export class ScheduledTaskSQSStrategy implements ScheduledTaskBaseStrategy {
+	public readonly options: ConsumerOptions;
+
+	private producer!: Producer;
+
+	public constructor(options: ConsumerOptions) {
+		this.options = options;
+	}
+
+	public connect() {
+		this.producer = Producer.create(this.options);
+
+		const consumer = Consumer.create({
+			...this.options,
+			// handleMessage requires a promise method as value. But we don't need to..
+			// eslint-disable-next-line @typescript-eslint/require-await
+			handleMessage: async (message) => void this._handleMessage(message),
+			// eslint-disable-next-line @typescript-eslint/require-await
+			handleMessageBatch: async (messages) => void this._handleBatch(messages)
```

Comment:
What? Why?

Reply:
@jurienhamaker can you answer these question?

Reply:
@feralheart nobody knows what he is saying